src/WebJobs.Script.Extensibility/ScriptBindingContext.cs

```diff
@@ -107,12 +114,32 @@ public ScriptBindingContext(JObject bindingMetadata)
         public TValue GetMetadataValue<TValue>(string name, TValue defaultValue = default(TValue))
         {
             JToken value = null;
-            if (Metadata.TryGetValue(name, StringComparison.OrdinalIgnoreCase, out value))
+            try
             {
-                return value.Value<TValue>();
+                if (Metadata.TryGetValue(name, StringComparison.OrdinalIgnoreCase, out value))
+                {
+                    return value.Value<TValue>();
+                }
+            }
+            catch (FormatException e)
+            {
+                throw new FormatException($"Invalid value specified for binding property '{name}' of type {PrettyTypeName(typeof(TValue))}.", e);
             }
 
             return defaultValue;
         }
+
+        private static string PrettyTypeName(Type t)
```

Comment:
We have an existing helper in the core SDK that does this same thing: https://github.com/Azure/azure-webjobs-sdk/blob/663a508e8a851629c26a51e7de3af36629dfd120/src/Microsoft.Azure.WebJobs.Host/TypeUtility.cs#L13

It only does it for Nullable types though to correct that issue. Other types are left as is.